x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,199 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
 		})
 	}
 }
+
+// TestFunctionalInternalExternalCLGauge is a functional test that covers more complex scenarios relating to distributing incentives through gauges
+// at the end of each epoch.
+//
+//
+// Testing strategy:
+// 1. Initialize variables.
+// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+// 3. Create external no-lock gauges for CL pools
+// 4. Create Distribution records to incentivize internal CL no-lock gauges
+// 5. let epoch 1 pass
+//  	- Check that incentive record has been correctly created and gauge has been correctly updated.
+// 		- all perpetual gauges must finish distributing records
```

Comment:
Might be good to mention that internal gauges are not supposed to be filled at this point